modules/configuration-api/src/main/java/org/apache/ignite/configuration/RootKey.java

```diff
@@ -38,8 +38,10 @@ public class RootKey<T extends ConfigurationTree<VIEWT, ?>, VIEWT> {
     /** Schema class for the root. */
     private final Class<?> schemaClass;
 
-    /** Marked with {@link InternalConfiguration}. */
-    private final boolean internal;
+    /** Marked with {@link ConfigurationExtension}. */
+    private final boolean privateExtension;
+
+    private final boolean publicExtension;
```

Comment:
Can you add javadoc here as well?

Reply:
UPD: I think it is enough to have a single field indicating if it is a public or private extension.